src/safe_thread_pool.h

```diff
@@ -0,0 +1,83 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "default_thread_pool.h"
+#include "logger.h"
+
+namespace vsag {
+
+class SafeThreadPool : public ThreadPool {
```

Comment:
why we need a safe thread pool?

Reply:
Similar to the allocator, use safe_thread_pool to manage the externally provided thread_pool and the internally created thread_pool.

Reply:
@LHT129 any suggestion?

Reply:
I agree the safe wrapper is used for threadpool behavior rather than the task